underfs/cephfs/src/main/java/alluxio/underfs/cephfs/CephFSUnderFileSystem.java

```diff
@@ -0,0 +1,792 @@
+  private static final int MAX_TRY = 5;
+
+  private CephMount mMount;
+
+  /**
+   * Factory method to constructs a new CephFS {@link UnderFileSystem} instance.
+   *
+   * @param ufsUri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for Hadoop
+   * @return a new CephFS {@link UnderFileSystem} instance
+   */
+  public static CephFSUnderFileSystem createInstance(
+      AlluxioURI ufsUri, UnderFileSystemConfiguration conf) throws IOException {
+    LOG.info("CephFS URI: {}", ufsUri.toString());
+
+    /*
+     * Create mount with auth user id
+     */
+    String userId = conf.get(PropertyKey.UNDERFS_CEPHFS_AUTH_ID);
+    LOG.info("CephFS config: {} = {}", PropertyKey.UNDERFS_CEPHFS_AUTH_ID, userId);
+    if (userId != null && userId.isEmpty()) {
+      userId = null;
+    }
+    CephMount mount = new CephMount(userId);
+
+    /*
+     * Load a configuration file if specified
+     */
+    String configfile = conf.get(PropertyKey.UNDERFS_CEPHFS_CONF_FILE);
+    LOG.info("CephFS config: {} = {}", PropertyKey.UNDERFS_CEPHFS_CONF_FILE, configfile);
+    if (configfile != null && !configfile.isEmpty()) {
+      mount.conf_read_file(configfile);
+    }
+
+    /*
+     * Parse and set Ceph configuration options
+     */
+    String configopts = conf.get(PropertyKey.UNDERFS_CEPHFS_CONF_OPTS);
+    LOG.info("CephFS config: {} = {}", PropertyKey.UNDERFS_CEPHFS_CONF_OPTS, configopts);
+    if (configopts != null && !configopts.isEmpty()) {
+      String[] options = configopts.split(";");
+      for (String option : options) {
+        String[] keyval = option.split("=");
+        if (keyval.length != 2) {
+          throw new IllegalArgumentException("Invalid Ceph option: " + option);
+        }
+        String k = keyval[0];
+        String v = keyval[1];
+        try {
+          mount.conf_set(k, v);
+        } catch (Exception e) {
+          throw new IOException("Error setting Ceph option " + k + " = " + v);
+        }
+      }
+    }
+
+    /*
+     * Set auth key
+     */
+    String key = conf.get(PropertyKey.UNDERFS_CEPHFS_AUTH_KEY);
+    if (key != null && !key.isEmpty()) {
+      mount.conf_set("key", key);
+    }
+
+    /*
+     * Set auth keyfile
+     */
+    String keyfile = conf.get(PropertyKey.UNDERFS_CEPHFS_AUTH_KEYFILE);
+    LOG.info("CephFS config: {} = {}", PropertyKey.UNDERFS_CEPHFS_AUTH_KEYFILE, keyfile);
+    if (keyfile != null && !keyfile.isEmpty()) {
+      mount.conf_set("keyfile", keyfile);
+    }
+
+    /*
+     * Set auth keyring
+     */
+    String keyring = conf.get(PropertyKey.UNDERFS_CEPHFS_AUTH_KEYRING);
+    LOG.info("CephFS config: {} = {}", PropertyKey.UNDERFS_CEPHFS_AUTH_KEYRING, keyring);
+    if (keyring != null && !keyring.isEmpty()) {
+      mount.conf_set("keyring", keyring);
+    }
+
+    /*
+     * Set monitor hosts
+     */
+    String monHost;
+    String host = ufsUri.getHost();
+    int port = ufsUri.getPort();
```

Comment:
can't we get `monHost` from `ufsUri.getAuthority`?
In this case do we still need to add `getHost` and `getPort`?